src/components/molecules/TabsContainer.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import TabMenu from './TabMenu';
+
+interface IProps {
+  className?: string;
+  children: Array<JSX.Element>;
```

Comment:
Wondering if there might be a better way to typecheck that the children must be Tab-components? As for the way I've implemented it I need to access the label-prop of the children to create the tablabels and map them to the correct content.

Reply:
Also check out this part

>We can use following types for annotating children:
ReactNode | ReactChild | ReactElement
object | {[key:string]:unknown} | MyModel
primitives string | number | boolean
Array<T> where T can be any of former
never | null | undefined ( null and undefined doesn't make much sense though )

You were using `JSX.Element` which derives from `React.ReactElement<any, any>`, which is why it worked.

Reply:
OK, almost done with my rant now, but this is from another section of the article.

> 18. Don’t use JSX.Element to annotate function/component return type or children/props

The reasoning:

> globals are bad ☝️💥
TypeScript supports locally scoped JSX to be able to support various JSX factory types and proper JSX type checking per factory. While current react types use still global JSX namespace, it’s gonna change in the future.
explicit types over generalized ones

Reply:
Great :100: